Test/Integration/Search/SearchTest.php

```diff
@@ -8,18 +8,34 @@
 
 class SearchTest extends TestCase
 {
-    public function testSearch()
+    /** @var Product */
+    protected $productIndexer;
+
+    /** @var Data */
+    protected $helper;
+
+    public function setUp(): void
     {
-        /** @var Product $indexer */
-        $indexer = $this->getObjectManager()->create(Product::class);
-        $indexer->executeFull();
+        parent::setUp();
 
+        $this->productIndexer = $this->objectManager->get(Product::class);
+        $this->helper = $this->getObjectManager()->create(Data::class);
+
+        $this->productIndexer->executeFull();
         $this->algoliaHelper->waitLastTask();
+    }
 
-        /** @var Data $helper */
-        $helper = $this->getObjectManager()->create(Data::class);
-        list($results, $totalHits, $facetsFromAlgolia) = $helper->getSearchResult('', 1);
+    public function testSearch()
+    {
+        list($results, $totalHits, $facetsFromAlgolia) = $this->helper->getSearchResult('', 1);
+        $this->assertNotEmpty($results);
+    }
 
+    public function testCategorySearch()
+    {
+        list($results, $totalHits, $facetsFromAlgolia) = $this->helper->getSearchResult('', 1, [
+            'facetFilters' => ['categoryIds:' . $this->assertValues->expectedCategory]
+        ]);
         $this->assertNotEmpty($results);
```

Comment:
Can we verify that the number of products retrieved with the category facet matches was is expected - perhaps against a collection through `CategoryRepository` and then verify parity with the Algolia index? i.e. counts should be identical. Keep in mind other settings such as whether included in menu etc'.